gpflux/layers/basis_functions/utils.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) 2021 The GPflux Contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module provides a set of common utilities for kernel feature decompositions.
+"""
+import tensorflow as tf
+import gpflow
+
+from typing import Tuple, Type
+from gpflow.base import DType, TensorType
+from gpflux.types import ShapeType
+
+
+RFF_SUPPORTED_KERNELS: Tuple[Type[gpflow.kernels.Stationary], ...] = (
+    gpflow.kernels.SquaredExponential,
+    gpflow.kernels.Matern12,
+    gpflow.kernels.Matern32,
+    gpflow.kernels.Matern52,
+)
+
+
+def _matern_number(kernel: gpflow.kernels.Kernel) -> int:
+    if isinstance(kernel, gpflow.kernels.Matern52):
+        p = 2
+    elif isinstance(kernel, gpflow.kernels.Matern32):
+        p = 1
+    elif isinstance(kernel, gpflow.kernels.Matern12):
+        p = 0
+    else:
+        raise NotImplementedError("Not a recognised Matern kernel")
```

Comment:
Minor: we *try* to use US spelling in code:
```suggestion
        raise NotImplementedError("Not a recognized Matern kernel")
```